notebook/services/kernels/kernelmanager.py

```diff
@@ -225,3 +256,42 @@ def record_activity(msg_list):
 
         kernel._activity_stream.on_recv(record_activity)
 
+    def initialize_culler(self):
+        """Start idle culler if 'cull_kernels_after_minutes' is greater than zero.
+
+        Regardless of that value, set flag that we've been here.
+        """
+        if not self._initialized_culler and self.cull_kernels_after_minutes > 0:
+            if self._culler_callback is None:
+                loop = IOLoop.current()
+                if self.kernel_culling_interval_seconds <= 0: #handle case where user set invalid value
+                    self.log.warn("Invalid value for 'kernel_culling_interval_seconds' detected (%s) - using default value (%s).",
+                        self.kernel_culling_interval_seconds, self.kernel_culling_interval_seconds_default)
+                    self.kernel_culling_interval_seconds = self.kernel_culling_interval_seconds_default
+                self._culler_callback = PeriodicCallback(
+                    self.cull_kernels, 1000*self.kernel_culling_interval_seconds, loop)
+                self.log.info("Culling kernels with idle durations > %s minutes at %s second intervals ...",
+                    self.cull_kernels_after_minutes, self.kernel_culling_interval_seconds)
+                self._culler_callback.start()
+
+        self._initialized_culler = True
+
+    def cull_kernels(self):
+        self.log.debug("Polling every %s seconds for kernels idle > %s minutes...",
+            self.kernel_culling_interval_seconds, self.cull_kernels_after_minutes)
+        for kId, kernel in self._kernels.items():
+            self.cull_kernel(kId, kernel)
+
+    def cull_kernel(self, kId, kernel):
+        activity = kernel.last_activity
+        name = kernel.kernel_name
+        self.log.debug("kId=%s, name=%s, last_activity=%s", kId, name, activity)
+        if activity is not None:
+            dtNow = utcnow()
+            #dtActivity = datetime.strptime(activity,'%Y-%m-%dT%H:%M:%S.%f')
+            dtIdle = dtNow - activity
```

Comment:
In Python, we try to use `_` to separate words, such as `kernel_id`, `idle_time` instead of `camelCase`